static/skin/index.js

```diff
@@ -142,10 +142,19 @@
         }
     }
 
-    function insertModal(button) {
+    async function getMagnetLink(downloadLink) {
+        const magnetUrl = downloadLink + '.magnet';
+        const magnetLink = await fetch(magnetUrl).then(response => {
+            return response.ok ? response.text() : '';
+        }).catch((error) => console.log(error));
+        return magnetLink;
+    }
+
+    async function insertModal(button) {
```

Comment:
Why do you need the `async` here? The fact that this function registers an `async` event listener doesn't make it `async` itself.

Reply:
Forgot to remove because an earlier version had `await getMagnetLink()` outside the listener, fixed now.